chapter.md

```diff
@@ -0,0 +1,735 @@
+  switch (action.type) {
+
+  case 'SAVE_FORM':
+    const lensForProp = lensPath(action.payload.path);
+    return assocPath(
+      action.payload.path,
+      merge(view(lensForProp, state), action.payload.value),
+      state
+    );
+
+  default:
+    return state;
+  }
+}
+```
+
+There is actually less going on here than first appears.
+Working from the outside in,
+if the action's type is *not* `SAVE_FORM`,
+then `formReducer` returns the original state without any changes.
+This means that we can safely chain this reducer together
+with others that handle other parts of our user interface:
+as long as they all use distinct keys for their actions,
+they will watch the state fly by without doing anything we don't want them to.
+
+If the action does have the right type,
+`formReducer` uses `lensPath` to figure out which part of the state needs to be updated,
+then uses `merge` and `assocPath` to create a copy of the state
+in which that element,
+*and only that element*,
+has a different value.
+The most important word in the previous sentence is "create":
+at this point,
+the state that was passed in is thrown away and a new one created.
+Some parts of the old state are recycled–`assocPath` does a shallow clone–but
+those are the parts that *weren't* changed,
+and if other reducers do their jobs properly,
+they will replace those parts rather than updating them in place
+when it's their turn to act.
+
+## Connecting the Dots
+
+It's now time to turn our attention to the interface
+that will trigger this state change
+and reflect any changes to the state triggered by other interface components.
+Since we're using Angular 2,
+we will define a class
+and use the `@Component` decorator to weld some metadata to it:
+
+```ts
+@Component({
+  selector: 'character-form',
+  template: require('./character-form.html')
+})
+export class CharacterForm {
+  @ViewChild(NgForm) ngForm: NgForm;
+  public characterForm;
+  private formSubs;
+
+  constructor(private ngRedux: NgRedux<IAppState>) {}
+}
+```
+
+There's a lot going on here,
+so we'll start with a high-level explanation and then dive into details.
+As the diagram shows,
+Angular will keep `ngForm` and `characterForm` in sync with each other
+using its own dark magic.
+In order to synchronize that with our state,
+we create a standard observer/observable connection
+to make `characterForm` watch the character portion of our Redux state.
+FIXME: I'm waving my hands a bit here...
+
+FIXME: diagram
+
+In more detail:
+
+*   The `@Component` decorator in the code above tells Angular that
+    this class is used to fill in `<character-form>` elements
+    in the `character-form.html` template snippet.
+*   Using the `@ViewChild` decorator on `ngForm`
+    gives this class an instance variable that watches a form.
+    We need this because we are going to subscribe to event notifications from that form
+    later on.
+*   The `characterForm` instance variable is our working copy of the form's state.
```

Comment:
this is actually the reference from the same object on the state, that will be automatically updated due the observable whenever the state changes